lib/magma/actions/action_error.rb

```diff
@@ -0,0 +1,15 @@
+class Magma
+  class ActionError
```

Comment:
A Struct could make a lot of sense here, but I'd recommend against subclassing it. It goes against the intended use of Structs

> ``` 
> Customer = Struct.new(:name, :address) do
>   def greeting
>     "Hello #{name}!"
>   end 
> end 
> Customer.new("Dave", "123 Main").greeting  #=> "Hello Dave!"
> ``` 
> 
> This is the recommended way to customize a struct. Subclassing an anonymous struct creates an extra anonymous class that will never be used.

https://ruby-doc.org/core-2.7.1/Struct.html

and it could lead to strange behavior down the road.

_I'm an expert on this topic because my wife [gave a talk](http://latazzajones.github.io/tashas_presentations/ruby_structs.html#/_) on it three years ago 😆_